sqlite.zig

```diff
@@ -1410,17 +1410,34 @@ pub const DynamicStatement = struct {
     // If however there are no name bind markers then the behaviour will revert to using the field index in the struct, and the fields order must be correct.
     fn bind(self: *Self, options: anytype, values: anytype) !void {
         const StructType = @TypeOf(values);
-        const StructTypeInfo = @typeInfo(StructType).Struct;
 
-        inline for (StructTypeInfo.fields) |struct_field, struct_field_i| {
-            const field_value = @field(values, struct_field.name);
+        switch (@typeInfo(StructType)) {
+            .Struct => |StructTypeInfo| {
+                inline for (StructTypeInfo.fields) |struct_field, struct_field_i| {
+                    const field_value = @field(values, struct_field.name);
 
-            const i = sqlite3BindParameterIndex(self.stmt, struct_field.name);
-            if (i >= 0) {
-                try self.bindField(struct_field.field_type, options, struct_field.name, i, field_value);
-            } else {
-                try self.bindField(struct_field.field_type, options, struct_field.name, struct_field_i, field_value);
-            }
+                    const i = sqlite3BindParameterIndex(self.stmt, struct_field.name);
+                    if (i >= 0) {
+                        try self.bindField(struct_field.field_type, options, struct_field.name, i, field_value);
+                    } else {
+                        try self.bindField(struct_field.field_type, options, struct_field.name, struct_field_i, field_value);
+                    }
+                }
+            },
+            .Pointer => |PointerTypeInfo| {
+                // TODO support other pointer types
+                std.debug.assert(PointerTypeInfo.size == .Slice);
```

Comment:
I'd prefer a switch on the size and explicitly using a `@compileError` if it isn't a `Slice`.

(also note that this assert doesn't work, it needs to be prefixed with `comptime` to actually trigger if you pass another pointer.